diff --git a/index.d.ts b/index.d.ts
index 95ebb17f1e78f456f2e634184b794b002132cbc9..7176d87d743498314168e15272d1572938fe6d52 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -9,7 +9,7 @@ export class AbortError extends Error {
 
 	@param message - An error message or a custom error.
 	*/
-	constructor(message: string | Error);
+	constructor(message: string | Error, options?: { cause?: any });
 }
 
 export type FailedAttemptError = {
diff --git a/index.js b/index.js
index da2c17f2809504894e322773c83d7ff95f7db556..20a7808a5143e27db20e48619bd9fd4109f9bbfc 100644
--- a/index.js
+++ b/index.js
@@ -2,7 +2,7 @@ import retry from 'retry';
 import isNetworkError from 'is-network-error';
 
 export class AbortError extends Error {
-	constructor(message) {
+	constructor(message, { cause } = {}) {
 		super();
 
 		if (message instanceof Error) {
@@ -11,6 +11,7 @@ export class AbortError extends Error {
 		} else {
 			this.originalError = new Error(message);
 			this.originalError.stack = this.stack;
+			this.originalError.cause = cause;
 		}
 
 		this.name = 'AbortError';
