diff --git a/index.d.ts b/index.d.ts
index aad3b32c11e2429313bf3c7ee400eb0dec6f195a..d47b2ff857240b84a2bb2a45c6a6b54f7c55febb 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -7,7 +7,7 @@ export class AbortError extends Error {
 
 	@param message - An error message or a custom error.
 	*/
-	constructor(message: string | Error);
+	constructor(message: string | Error, options?: { cause?: any });
 }
 
 export type RetryContext = {
diff --git a/index.js b/index.js
index 1a7deb776667e880c0717e81ddb22a9d81a96590..bbc9937c5b232b7acc9cb92b9357f1bd365cb53c 100644
--- a/index.js
+++ b/index.js
@@ -33,7 +33,7 @@ function validateNumberOption(name, value, {min = 0, allowInfinity = false} = {}
 }
 
 export class AbortError extends Error {
-	constructor(message) {
+	constructor(message, { cause } = {}) {
 		super();
 
 		if (message instanceof Error) {
@@ -42,6 +42,7 @@ export class AbortError extends Error {
 		} else {
 			this.originalError = new Error(message);
 			this.originalError.stack = this.stack;
+			this.originalError.cause = cause;
 		}
 
 		this.name = 'AbortError';
